pkg/controller/datavolume-controller.go

```diff
@@ -2823,14 +2823,27 @@ func (r *DatavolumeReconciler) detachPvcDeleteDv(pvc *corev1.PersistentVolumeCla
 }
 
 func updatePvcOwnerRefs(pvc *corev1.PersistentVolumeClaim, dv *cdiv1.DataVolume) {
-	refs := pvc.OwnerReferences
-	for i, r := range refs {
-		if r.UID == dv.UID {
-			pvc.OwnerReferences = append(refs[:i], refs[i+1:]...)
+	var refs []metav1.OwnerReference
+	for _, pvcOwnerRef := range pvc.OwnerReferences {
+		if pvcOwnerRef.UID != dv.UID {
+			refs = append(refs, pvcOwnerRef)
 			break
 		}
 	}
-	pvc.OwnerReferences = append(pvc.OwnerReferences, dv.OwnerReferences...)
+	for _, dvOwnerRef := range dv.OwnerReferences {
+		refExists := false
+		for _, pvcOwnerRef := range pvc.OwnerReferences {
+			if dvOwnerRef.UID == pvcOwnerRef.UID {
+				refExists = true
+				break
+			}
+		}
+		if !refExists {
+			dvOwnerRef.BlockOwnerDeletion = nil
```

Comment:
maybe a comment above this with the admission plugin name and short explanation that we're taking ownerrefs that can be anything in the world?

Reply:
Added in the PR description. Looks ok?

Reply:
Sure, adding.